spec/design/skel/FaultHandler_H.thy

```diff
@@ -13,15 +13,14 @@ imports
   ArchFaultHandler_H
 begin
 
-context begin interpretation Arch .
-requalify_consts
-  syscallMessage
-  fromVPtr
-  exceptionMessage
-  debugPrint
+arch_requalify_consts (H)
   makeArchFaultMessage
   handleArchFaultReply
-end
+
+(* FIXME arch-split: clobbers generic with arch version, no disambiguation via global. *)
+arch_requalify_consts (aliasing, H)
+  syscallMessage
+  exceptionMessage
```

Comment:
Do we really want to fix this (i.e. should it be FIXME), or should we just leave a warning comment here? It does seem like the proofs are expecting the current state, so we probably don't really want to change this.

Reply:
The issue I had for this one is that the *current*, pre-arch-split proofs expecting it doesn't mean that it's the right way around, especially minus any disambiguation. I guess the current proofs do indicate they want this as you say. Will double check.

Reply:
This is very well spotted, and made me go look at the other FIXME. What's happening is that we requalified the Arch version in ASpec (e.g. from AARCH64 or AARCH64_A), and now we're requalifying the Haskell version with the same name (AARCH64_H).

When I looked at the *other* FIXME though, it turned out that this was also the case except for `sanitiseRegister` and `getSanitiseRegisterInfo` ... oh man, it turns out that we imported *only* the generic const for those two, then the Arch const+def, then requalified the Arch const to generic (clobbering the generic const), then used type-compatibility to have generic imported definitions that mention those constants end up using the Arch ones instead!
I guess someone didn't want to import a generic constant that was essentially defined to be the same as the Arch constant, but then still left the original generic constants floating around. Super confusing. So at least to alleviate confusion, I removed the imports of the unused constants, which also meant no aliasing warning.

Reply:
Cool, glad that this untangled parts of it.